Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs

```diff
@@ -1122,3 +1152,54 @@ remove (personId, _) = do
   Esq.runTransaction $ do
     QPerson.updateAlternateMobileNumberAndCode driver
   return Success
+
+getDriverPayments :: (EsqDBReplicaFlow m r, EsqDBFlow m r, EncFlow m r, CacheFlow m r) => (Id SP.Person, Id DM.Merchant) -> Maybe Day -> Maybe Day -> Maybe Int -> Maybe Int -> m [DriverPaymentHistoryResp]
+getDriverPayments (_, merchantId_) mbFrom mbTo mbLimit mbOffset = do
+  let limit = min maxLimit . fromMaybe defaultLimit $ mbLimit -- TODO move to common code
+      offset = fromMaybe 0 mbOffset
+      defaultFrom = fromMaybe (fromGregorian 2020 1 1) mbFrom
+  nowUtc <- getCurrentTime
+  transporterConfig <- CQTC.findByMerchantId merchantId_ >>= fromMaybeM (TransporterConfigNotFound merchantId_.getId)
+  let now = utctDay $ getLocalTime nowUtc transporterConfig.timeDiffFromUtc
+      from = fromMaybe defaultFrom mbFrom
+      to = fromMaybe now mbTo
+  let windowStartTime = UTCTime from 0
+      windowEndTime = addUTCTime (86399 + transporterConfig.driverPaymentCycleDuration) (UTCTime to 0)
```

Comment:
why this?

Reply:
Because today's rides might have endTime of `today + cycleDuration`. Also, I want to start counting from last timestamp(11:59 PM today) so that no ride is missed @hkmangla